controllers/front/ProductController.php

```diff
@@ -197,7 +197,7 @@ public function init()
                 $this->errors[] = $this->trans('You do not have access to this product.', [], 'Shop.Notifications.Error');
                 $this->setTemplate('errors/forbidden');
             } else {
-                if (!$isAssociatedToProduct && $isPreview) {
+                if (!$isAssociatedToProduct && $isPreview && !$this->ajax) {
```

Comment:
What if it is ajax then?

Reply:
If it's in ajax, we bypass the error because the `adtoken` for preview is not passed to ajax requests.

Reply:
It would be more undeestandable for me if you put the check
Tools::getValue('adtoken') == Tools::getAdminToken(
                    'AdminProducts'
                    . (int) Tab::getIdFromClassName('AdminProducts')
                    . (int) Tools::getValue('id_employee')
                )
                
in a variable and set it to true if isAjax

WDYT ?

Reply:
@sowbiba No because `$isAssociatedToProduct` is used in other places of the code with the not same condition.